src/controllers/users.controller.js

```diff
@@ -0,0 +1,79 @@
+const usersServices = require('../services/users/users.services');
+
+const getUsers = (_, res) => {
+  return res.status(200).json(usersServices.getUsers());
+};
+
+const getUserById = (req, res) => {
+  const { id } = req.params;
+
+  const userId = Number(id);
+
+  if (!userId || isNaN(userId)) {
+    return res.sendStatus(400);
+  }
+
+  const user = usersServices.getUserById(userId);
+
+  if (!user) {
+    res.sendStatus(404);
+
+    return;
+  }
+  res.send(user);
+
+  return !user ? res.sendStatus(404) : res.status(200).json(user);
+};
+
+const saveUser = (req, res) => {
+  const { name } = req.body;
+
+  if (!name || typeof name !== 'string') {
+    return res.sendStatus(400);
+  }
+
+  const newUser = usersServices.saveUser(name);
+
+  return res.status(201).json(newUser);
+};
+
+const removeUser = (req, res) => {
+  const { id } = req.params;
+
+  const userId = Number(id);
+  const user = usersServices.getUserById(userId);
+
+  if (!user) {
+    return res.sendStatus(404);
+  }
+
+  usersServices.removeUser(userId);
+
+  res.sendStatus(204);
+};
+
+const updateUser = (req, res) => {
+  const { id } = req.params;
+  const { name } = req.body;
+  const userId = Number(id);
+
+  if (!userId || isNaN(userId) || typeof name !== 'string') {
```

Comment:
The condition `!userId || isNaN(userId) || typeof name !== 'string'` can be simplified by removing `!userId` since `isNaN(userId)` already covers invalid number cases.